ElectronNET.Host/api/app.ts

```diff
@@ -1,13 +1,20 @@
 let isQuitWindowAllClosed = true, electronSocket;
+let appWindowAllClosedEventId;
 export = (socket: SocketIO.Socket, app: Electron.App) => {
     electronSocket = socket;
-    // Quit when all windows are closed.
+
+    // By default, quit when all windows are closed
     app.on('window-all-closed', () => {
         // On macOS it is common for applications and their menu bar
         // to stay active until the user quits explicitly with Cmd + Q
-        if (process.platform !== 'darwin' &&
-            isQuitWindowAllClosed) {
+        if (process.platform !== 'darwin' && isQuitWindowAllClosed) {
             app.quit();
+        } else if (!!appWindowAllClosedEventId) {
```

Comment:
Normally no two exclamation points necessary... reads more complicated... JavaScript already understands what you want :) it only goes through if it has a valid value.. change and test with `else if (appWindowAllClosedEventId) {`

Reply:
I've gotten into the habit of `!!` simply due to this:

https://brianflove.com/2014-09-02/whats-the-double-exclamation-mark-for-in-javascript/

`var foo = ''` will cause `if (foo) { console.log('foo is true!'); }` to output 'foo is true'.  In reality, I want to treat an empty string as false.  Thus, a `!!` will convert the var to a bool and then take the inverse.  `if (!!foo) { console.log('foo is true!'); }` will not output a value.  This means I don't have to do any extra checking of undefined, null, or empty string.

Let me know if you'd rather me swap it to the single `!`.  Happy to do so.